server/scripts/sendEmailBlast.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-console */
+require('dotenv').config({ path: '../.env' });
+const axios = require('axios');
+const { dbClient } = require('../db/db');
+
+const CHES_HOST = process.env.CHES_HOST || 'https://ches-dev.apps.silver.devops.gov.bc.ca';
+const AUTH_URL =
+  process.env.AUTH_URL ||
+  'https://dev.oidc.gov.bc.ca/auth/realms/jbd6rnxw/protocol/openid-connect/token';
+const failedSends = [];
+async function authenticateChes() {
+  const params = new URLSearchParams();
+  params.append('grant_type', 'client_credentials');
+
+  const config = {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    auth: { username: process.env.CHES_CLIENT_ID, password: process.env.CHES_CLIENT_SECRET }, // service client ID and secret
+  };
+  try {
+    const response = await axios.post(AUTH_URL, params, config);
+    return response.data.access_token;
+  } catch (error) {
+    console.log({ '@@@@': error });
+    return error;
+  }
+}
+function createPayload(recipient, uuid) {
+  return {
+    from: 'noreply@hcapparticipants.gov.bc.ca',
+    to: [recipient],
+    subject: 'Are you still interested in the Health Career Access Program?',
+    bodyType: 'html',
+    body: `
+        <body>
+            If you would still like to participate in the program, <br/>
+            please confirm your interest by clicking on the link below. Clicking on the link will reconfirm your interest in the program to ensure you remain visible to eligible employers.
+            <br/><br/>
+            <button>
+                <a href="${`${process.env.BACKEND_URL}/confirm-interest?token=${uuid}`}">Confirm Interest</a>
+            </button>
+            <br/><br/>
+            If you no longer wish to participate or be considered for the Health Career Access Program, please email us with the subject line WITHDRAW to <b>HCAPInfoQuery@gov.bc.ca</b>
+            <br/><br/>
+            <button>
+                <a href="mailto:HCAPInfoQuery@gov.bc.ca?subject=WITHDRAW&body=Hello,%0d%0aPlease%20withdraw%20the%20expression(s)%20of%20interest%20for%20the%20sender%20of%20this%20email.%0d%0aI%20am%20no%20longer%20interested%20in%20participating%20in%20the%20Health%20Career%20Access%20Program.">Withdraw from HCAP</a> 
+            </button>
+        </body>
+        
+        `,
+  };
+}
+
+const config = (token) => ({
+  headers: {
+    authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json',
+  },
+});
+
+async function getEmailBlock(index, block) {
+  return dbClient.db.query(
+    `SELECT email_address , otp FROM confirm_interest LIMIT ${block} OFFSET ${index}`
```

Comment:
Just a conversation point but I'd prefer to use the massive functions if possible, this is probably all subjective though